apps/src/templates/VersionHistory.jsx

```diff
@@ -0,0 +1,92 @@
+var React = require('react');
+var VersionRow = require('./VersionRow.jsx');
+var sourcesApi = require('../clientApi')('sources');
+
+/**
+ * A component for viewing project version history.
+ */
+module.exports = React.createClass({
+
+  getInitialState: function () {
+    return {
+      versions: null,
+      statusMessage: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    sourcesApi.ajax('GET', 'main.json/versions', this.onVersionListReceived, this.onAjaxFailure);
```

Comment:
There's no problem with dots in the URL path in general, e.g. `http://www.example.com/api/1.0.2/base/index.html`.  Is it bad to do this in a REST API?